validator/client/validator_propose.go

```diff
@@ -61,6 +73,12 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 		Signature: sig,
 	}
 
+	history = SetProposedForEpoch(history, epoch)
+	if err := v.db.SaveProposalHistory(ctx, pubKey[:], history); err != nil {
+		log.WithError(err).Error("Failed to save updated proposal history")
```

Comment:
Just quitting if there is a issue saving seems like a flawed pattern, it's saved before the block is proposed to make sure it isn't proposed when it shouldn't be, but how else can we handle this case?